File: sdk-ruby/lib/wave/chapters.rb
# frozen_string_literal: true

module Wave
  # Chapters — AI chapter detection and video segmentation
  class Chapters
    def initialize(client)
      @client = client
    end

    # List chapters for a video (operationId: listChapters, GET /videos/{videoId}/chapters).
    def list(video_id)
      @client.request("GET", "/videos/#{video_id}/chapters")
    end

    # Create a chapter (operationId: createChapter, POST /videos/{videoId}/chapters).
    def create(video_id, body)
      @client.request("POST", "/videos/#{video_id}/chapters", body: body)
    end

    # Start AI chapter detection (operationId: detectChapters, POST /videos/{videoId}/chapters/detect).
    def detect(video_id, body)
      @client.request("POST", "/videos/#{video_id}/chapters/detect", body: body)
    end

  end
end

File: sdk-rust/wave/src/chapters.rs
//! Chapters — AI chapter detection and video segmentation
#![allow(clippy::all)]
use wave_core::{models, Client, Error};

/// Service handle for the Chapters product.
pub struct Chapters<'a> {
    pub(crate) http: &'a Client,
}

impl<'a> Chapters<'a> {
    /// List chapters for a video (operationId: listChapters, GET /videos/{videoId}/chapters).
    pub fn list(&self, video_id: &str) -> Result<serde_json::Value, Error> {
        let path = "/videos/{videoId}/chapters".to_string().replace("{videoId}", video_id);
        self.http.request("GET", &path, &[], None::<&serde_json::Value>)
    }

    /// Create a chapter (operationId: createChapter, POST /videos/{videoId}/chapters).
    pub fn create(&self, video_id: &str, body: &models::ChapterCreate) -> Result<models::Chapter, Error> {
        let path = "/videos/{videoId}/chapters".to_string().replace("{videoId}", video_id);
        self.http.request("POST", &path, &[], Some(body))
    }

    /// Start AI chapter detection (operationId: detectChapters, POST /videos/{videoId}/chapters/detect).
    pub fn detect(&self, video_id: &str, body: &models::ChapterDetectRequest) -> Result<models::DetectionJob, Error> {
        let path = "/videos/{videoId}/chapters/detect".to_string().replace("{videoId}", video_id);
        self.http.request("POST", &path, &[], Some(body))
    }

}